pywbem/_cim_operations.py

```diff
@@ -2060,39 +2133,53 @@ def infer_embedded_object(obj):
 
         return (returnvalue, output_params)
 
-    def _iparam_namespace_from_namespace(self, obj):
+    def _iparam_namespace_from_namespace(self, namespace):
         # pylint: disable=invalid-name,
         """
-        Determine the namespace from a namespace string, or `None`. The
-        default namespace of the connection object is used, if needed.
+        Determine the namespace from a namespace parameter, for use as an
+        argument to imethodcall().
 
-        Return the so determined namespace for use as an argument to
-        imethodcall().
-        """
+        Slashes at begin and end of namespace are stripped.
 
-        if isinstance(obj, six.string_types):
-            namespace = obj.strip('/')
-        elif obj is None:
-            namespace = obj
+        If namespace is None, the default namespace of the connection object
+        is used.
+
+        Returns:
+           The so determined namespace string.
+
+        Raises:
+          TypeError - namespace has an invalid type
+        """
+        if isinstance(namespace, six.string_types):
```

Comment:
We test for the / chars here, and I assume other places in client.  However, we also check all over the mock code. Not for this PR, but we should probably relook at that and clean it up so that there is only one check somewhere or none because the client does it.

Reply:
NOISSUE - This normalization you are commenting on is invoked at the first level in the WBEMConnection operations that is common for both the CIM-XML part (for parsing the namespace into its parts), and for the mocker. So the mocker already gets a normalized namespace, that is good to be a key in a case-insensitive dictionary. If the mock support strips again, that is not needed, and can be verified by having appropriate testcases.